web_infrastructure/jenkins_plugin.py

```diff
@@ -0,0 +1,753 @@
+#!/usr/bin/python
+# encoding: utf-8
+
+# (c) 2016, Jiri Tyr <jiri.tyr@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from pwd import getpwnam
+import base64
+import hashlib
+import json
+import os
+import tempfile
+import time
+
+
+DOCUMENTATION = '''
+---
+module: jenkins_plugin
+author: Jiri Tyr (@jtyr)
+version_added: '2.1'
+short_description: Add or remove Jenkins plugin
+description:
+  - Ansible module which helps to manage Jenkins plugins.
+
+options:
+  jenkins_group:
+    required: false
+    default: jenkins
+    description:
+      - Name of the Jenkins group on the OS.
+  jenkins_home:
+    required: false
+    default: /var/lib/jenkins
+    description:
+      - Home directory of the Jenkins user.
+  jenkins_user:
+    required: false
+    default: jenkins
+    description:
+      - Name of the Jenkins user on the OS.
+  name:
+    required: true
+    description:
+      - Plugin name.
+  params:
+    required: false
+    default: None
+    description:
+      - Option used to allow the user to overwrite any of the other options. To
+        remove an option, set the value of the option to C(null).
+  state:
+    required: false
+    choices: [absent, present, pinned, unpinned, enabled, disabled]
+    default: present
+    description:
+      - Desired plugin state.
+  timeout:
+    required: false
+    default: None
+    description:
+      - Server connection timeout in secs.
+  updates_expiration:
```

Comment:
Is this the correct name to use? seems like the apt module uses cache_valid_time or something like that. Is there consistency amongst modules out there?